torch/csrc/jit/python_interpreter.cpp

```diff
@@ -56,30 +55,25 @@ Operation createPythonOperation(Node* op_) {
     py::object py_outputs(func(*py_inputs));
 
     auto num_outputs = op->outputs().size();
-    auto addOutput = [&](py::handle entry) {
-      if (!THPVariable_Check(entry.ptr())) {
-        throw std::runtime_error(
-            "Function application returned a non-Variable output");
-      }
-      THPVariable* var = (THPVariable*)entry.ptr();
-      auto cdata = var->cdata;
-      stack.push_back(std::move(cdata));
+    auto addOutput = [&](py::handle entry, size_t pos) {
+      stack.push_back(returnToIValue(pos, op->outputs().at(pos)->type(), entry));
     };
 
     if (!PyTuple_Check(py_outputs.ptr())) {
```

Comment:
Do we treat the `-> Tuple[int, Tensor]` schema differently from `-> (int, Tensor)`? If yes, I think that we should branch on `num_outputs` instead of checking if the return value is a tuple. This would be a good test too.

Reply:
yeah, that is a better test. Ill make that change.